c/omni_lock.c

```diff
@@ -137,41 +137,27 @@ bool is_memory_enough(mol_seg_t seg, const uint8_t *cur, uint32_t len) {
 // <identity, 21 bytes> <omni_lock args>
 // <omni_lock flags, 1 byte>  <OMNI cell type id, 32 bytes, optional> <ckb/udt
 // min, 2 bytes, optional> <since, 8 bytes, optional>
-int parse_args(ArgsType *args) {
+int parse_args(ScriptType script, ArgsType *args) {
   int err = 0;
-  uint8_t script[SCRIPT_SIZE];
-  uint64_t len = SCRIPT_SIZE;
-  err = ckb_checked_load_script(script, &len, 0);
-  CHECK(err);
-
-  mol_seg_t script_seg;
-  script_seg.ptr = script;
-  script_seg.size = (mol_num_t)len;
-
-  mol_errno mol_err = MolReader_Script_verify(&script_seg, false);
-  CHECK2(mol_err == MOL_OK, ERROR_ENCODING);
 
-  mol_seg_t args_seg = MolReader_Script_get_args(&script_seg);
-  mol_seg_t seg = MolReader_Bytes_raw_bytes(&args_seg);
-
-  uint8_t *cur = seg.ptr;
+  // TODO: do we need to validate Script structure here?
```

Comment:
The script structure can be safely used without validation. It's directly from syscall. If this structure is wrong, the ckb consensus is in trouble.